rfc/src/rfcs/0007-global-conditions.md

```diff
@@ -0,0 +1,111 @@
+- **Feature Request Issue:** <https://github.com/model-checking/kani/issues/2525>
+- **RFC PR:** <https://github.com/model-checking/kani/pull/2516>
+- **Status:** *Unstable*
+- **Version:** 0
+- **Proof-of-concept:** <https://github.com/model-checking/kani/pull/2532>
+
+-------------------
+
+## Summary
+
+A new section in Kani's output to summarize the status of properties that depend on other properties. We use the term *global conditions* to refer to such properties. 
+
+## User Impact
+
+The addition of new options that affect the overall verification result depending on certain property attributes demands some consideration.
+In particular, the addition of a new option to fail verification if there are uncoverable (i.e., unsatisfiable or unreachable) `cover` properties (requested in [#2299](https://github.com/model-checking/kani/issues/2299)) is posing new challenges to our current architecture and UI.
+
+This concept isn't made explicit in Kani, but exists in some ways.
+For example, the `kani::should_panic` attribute is a global condition because it can be described in terms of other properties (checks).
+The request in [#2299](https://github.com/model-checking/kani/issues/2299) is essentially another global conditions, and we may expect more to be requested in the future.
+
+In this RFC, we propose a new section in Kani's output focused on reporting global conditions.
+The goal is for users to receive useful information about hyperproperties without it becoming overwhelming.
+This will help users to understand better options that are enabled through global conditions and ease the addition of such options to Kani.
+
+## User Experience
+
+**The output will refer to properties that depend on other properties as "global conditions"**, which is a simpler term.
+The options to enable different global conditions will depend on a case-by-case basis[^enable-options].
+
+The main UI change in this proposal is a new `GLOBAL CONDITIONS` section that **won't be printed if no global conditions have been enabled**.
+This section will only appear in Kani's default output after the `RESULTS` section (used for individual checks) and have the format:
+
+```
+GLOBAL CONDITIONS:
+ - `<name>`: <status> (<reason>)
+ - `<name>`: <status> (<reason>)
+ [...]
+```
+
+where:
+  - `<name>` is the name given to the global condition.
+  - `<status>` is the status determined for the global condition.
+  - `<reason>` is an explanation that depends on the status of the global condition.
+
+For example, let's assume we implement the option requested in [#2299](https://github.com/model-checking/kani/issues/2299).
+A concrete example of this output would be:
+
+```
+GLOBAL CONDITIONS:
+ - `fail_uncoverable`: SUCCESS (all cover statements were satisfied as expected)
+```
+
+A `FAILED` status in any enabled global condition will cause verification to fail.
+In that case, the overall verification result will point out that one or more global conditions failed, as in:
+
+```
```

Comment:
Is it possible to list the basic conditions that failed to explain why a global condition failed ?